drivers/serial/eusart_gecko.c

```diff
@@ -0,0 +1,571 @@
+/*
+ * Copyright (c) 2024, Yishai Jaffe
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_gecko_eusart
+
+#include <errno.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/clock_control_silabs.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/irq.h>
+#include <zephyr/pm/device.h>
+#include <em_eusart.h>
+#include <em_cmu.h>
+#include <soc.h>
+
+
+#define GECKO_UART_DEFAULT_BAUDRATE	115200
+#define GECKO_UART_DEFAULT_PARITY	UART_CFG_PARITY_NONE
+#define GECKO_UART_DEFAULT_STOP_BITS	UART_CFG_STOP_BITS_1
+#define GECKO_UART_DEFAULT_DATA_BITS	UART_CFG_DATA_BITS_8
```

Comment:
Same comment here regarding the defaults as I had in your other PR. The existence of them has (IMO) the risk of obfuscating what the actual used values are.